### ISSUE TITLE
Azure AD : hydrate function listAdSignInReports failed with panic runtime error

### ISSUE THREAD

**Issue by Bobaero** [turbot/steampipe-plugin-azuread#89]
**Describe the bug**
I'm trying to request Azure AD Sign In Reports to get the last signin time for my users, and my request always failed on this error :
`> select user_principal_name, max(created_date_time) from azuread_sign_in_report group by user_principal_name;
2022-10-21 12:44:28.070 UTC [TRACE] steampipe: resolveQuery select user_principal_name, max(created_date_time) from azuread_sign_in_report group by user_principal_name; args <empty>
2022-10-21 12:44:28.072 UTC [TRACE] steampipe: ensureSessionSearchPath
2022-10-21 12:44:28.072 UTC [TRACE] steampipe: updated the required search path to "public","azuread","steampipe","internal"
2022-10-21 12:44:28.072 UTC [TRACE] steampipe: session search path is already correct - nothing to do
Error: rpc error: code = Internal desc = hydrate function listAdSignInReports failed with panic runtime error: invalid memory address or nil pointer dereference (SQLSTATE HV000)`

**Steampipe version (`steampipe -v`)**
v0.16.4

**Plugin version (`steampipe plugin list`)**
v0.8.0

**To reproduce**
Log to your Azure AD : az login then use the query `select user_principal_name, max(created_date_time) from azuread_sign_in_report group by user_principal_name;`

**Expected behavior**
return the last signin time for every user on the azure ad reports.

**Additional context**
n/a

**Comment by cbruno10** [comment]
@Bobaero In addition to what @Subhajit97 had mentioned above, can you please also share:
- Does querying other tables work? For instance, the `azuread_user` and the `azuread_group` tables.
- What permissions does your service principal have in the AD? For reference, the permissions we recommend are listed in the **Permissions** row under [Credentials](https://hub.steampipe.io/plugins/turbot/azuread#credentials).
- How are you authenticating to Azure AD? Can you please share your `~/.steampipe/config/azuread.spc` file, with any sensitive info removed?

Thanks!

**Comment by Bobaero** [comment]
Hi @Subhajit97, @cbruno10 ,

Thanks for your fast answers. I've done your test :

- `select user_principal_name, created_date_time from azuread_sign_in_report limit 50` : This query work without throwing any error.
- `select * from azuread_user` throw an error : Error: rpc error: code = Internal desc = hydrate function listAdUsers failed with panic runtime error: invalid memory address or nil pointer dereference (SQLSTATE HV000)
- But `select * from azuread_user **limit 50**` work
- I'm logging in with the "az login --allow-no-subscriptions" command in my VSL 2.0 environment. So i've made no modification of the azuread.spc file
- And my user is a full administrator account.

Thanks again !

**Comment by cbruno10** [comment]
Hi @Bobaero , can you please try restarting the Steampipe service, and in the new session, turn off the cache, e.g., `.cache off`, and then try the queries again? We've done some local testing and found that sometimes caching the credentials causes queries to fail.

**Comment by Bobaero** [comment]
I've tried relaunching my WSL, login to AZ, turning off the cache but i've got the same issues (work with limit but not without).

I'm not sure i've restart steampipe service tho.

**Comment by cbruno10** [comment]
@Bobaero Can you please try running the following commands:
```
steampipe service stop --force
steampipe query
```

And then once inside the session:
```
.cache off
select id from azuread_user;
select id from azuread_user limit 50;
select id from azuread_group;
select id from azuread_group limit 10;
select user_principal_name, max(created_date_time) from azuread_sign_in_report group by user_principal_name;
```

Can you then please share the full terminal output from when you run `steampipe service stop --force`, including the query results (with any sensitive info removed/hidden) in this issue? Thanks!

**Comment by Subhajit97** [comment]
@Bobaero this bug should now be fixed in the new `AzureAD` plugin release [0.8.1](https://github.com/turbot/steampipe-plugin-azuread/blob/main/CHANGELOG.md#v081-2022-10-21). Let us know if you still see this error after updating; thanks!

**Comment by Bobaero** [comment]
Hi @Subhajit97,

Yes, this new version resolve my error.

Many thanks !
